sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1199,6 +1199,49 @@ case class Substring(str: Expression, pos: Expression, len: Expression)
 }
 
 /**
+ * Returns the rightmost n characters from the string.
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(str, len) - Returns the rightmost `len`(`len` can be string type) characters from the string `str`,if `len` is less or equal than 0 the result is ``.",
+  extended = """
+    Examples:
+      > SELECT _FUNC_('Spark SQL', 3);
+       SQL
+  """)
+// scalastyle:on line.size.limit
+case class Right(str: Expression, len: Expression, child: Expression) extends RuntimeReplaceable {
+  def this(str: Expression, len: Expression) = {
+    this(str, len, Substring(str, If(LessThanOrEqual(len, Literal(0)),
+      Literal(Integer.MAX_VALUE), UnaryMinus(len)), len))
```

Comment:
`right(null, -10)`
I agree with you, but , for this test case, there is a problem:
Which we expected is `null`,but it is an empty string